openquake/hazardlib/shakemap/parsers.py

```diff
@@ -160,16 +160,31 @@ def get_array_usgs_xml(kind, grid_url, uncertainty_url=None):
     """
     Read a ShakeMap in XML format from the local file system
     """
-    grid_url = path2url(grid_url)
-
-    if uncertainty_url is None:
-        with urlopen(grid_url) as f:
-            return get_shakemap_array(f)
-    else:
-        with urlopen(grid_url) as f1, urlextract(
-                path2url(uncertainty_url), 'uncertainty.xml') as f2:
+    try:
+        grid_url = path2url(grid_url)
+        if uncertainty_url is None:
+            if grid_url.endswith('.zip'):
+                # see if both are in the same file, naming must be correct
+                try:
+                    with urlextract(grid_url, 'grid.xml') as f1, urlextract(
+                            path2url(grid_url),
+                            'uncertainty.xml') as f2:
+                        return get_shakemap_array(f1, f2)
+                except FileNotFoundError:
+                    pass
```

Comment:
Perhaps a warning saying that the grid file is missing?

Reply:
If leaving out the engine doesn't even go that far since already oqvalidation.is_valid_shakemap returns an error.
 
```ValueError: Expected parameters ['kind', 'grid_url', 'uncertainty_url'] in shakemap_uri, got ['kind', 'grid_url']```

The case where no uncertainty grid is specified only happens in "tests" at the moment.

Reply:
yes, will fix it, just saw that there is another problem, let me fix that quickly

Reply:
I added the warning. Also it was not possible to have 
```
shakemap_uri = {
     "kind": "usgs_xml",
     "grid_url": "shakemap_files/grid.zip"}
```
Because oqvalidation raised an error. Fixed that too now.